### PR TITLE
Sockets - Grace

### PR DESCRIPTION
**Pull request by gracemshea**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model is communicating with the database. It can carry out all the actions... adding a task, updating it, destroying it etc. It can find and list tasks.
Describe in your own words what the Controller is doing in Rails | It's the middle man of sorts. It's getting info from the model and sending it to display on the views.
Describe in your own words what the View is doing in Rails | The view displays forms and data, and reacts to user inputs.
Describe an edge-case controller test you wrote | One of my edge cases ensures that the user is redirected if they request to edit a task that doesn't exist.
What is the purpose of using strong params? (i.e. the params method in the controller) | To DRY up the code, and control the data that is sent to the controller.
How are Rails migrations related to Rails models? | Migrations change what the model represents. Attributes are like the columns of the model/table. You can add new attributes with a migration.
Describe one area of Rails that are still unclear on | I feel like Rails is super magical, and I don't like magic. So much happening under the hood, I don't even know how to pinpoint what I want to know...

### CODE REVIEW

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -105,24 +110,84 @@
...
+      after_task = Task.find_by(id: task.id)
+      expect(after_task).must_be_nil
+    end
   end
 
-  # Complete for Wave 4
   describe "toggle_complete" do
-    # Your tests go here
+    it "can mark an incomplete task complete without changing anything else" do
+      # Arrange
```
Comment (droberts-sea) [inline test/controllers/tasks_controller_test.rb:168]: Two missing cases here:
- Does it go the other direction (complete -> incomplete)?
- What happens if the task doesn't exist?

**Comment by droberts-sea** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update | yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | mostly
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | You're off to a strong start on Rails so far! I especially like your styling work, and the idea of separating complete and incomplete tasks into separate lists. Keep up the hard work!
